doc/source/scripts/gen_format_overview_classes.py

```diff
@@ -7,115 +7,167 @@
 """
 
 from collections import defaultdict
+from typing import Any, Iterable, Literal, Type, TypeVar
 
+import base
 from base import TableWriter
 from core import DESCRIPTIONS
 from MDAnalysis import _CONVERTERS, _PARSERS, _READERS, _SINGLEFRAME_WRITERS
 
-FILE_TYPES = defaultdict(dict)
+HANDLER_T = Literal[
+    "Coordinate reader", "Coordinate writer", "Topology parser", "Converter"
+]
 
-for clstype, dct in (
+PAIR_T = tuple[HANDLER_T, dict[str, Any]]
+
+handlers: Iterable[PAIR_T] = [
     ("Coordinate reader", _READERS),
     ("Coordinate writer", _SINGLEFRAME_WRITERS),
     ("Topology parser", _PARSERS),
     ("Converter", _CONVERTERS),
-):
-    for fmt, klass in dct.items():
-        if fmt in ("CHAIN", "MEMORY", "MINIMAL", "NULL"):
+]
+
+FILE_TYPES: dict[str, dict[HANDLER_T, Any]] = defaultdict(dict)
+
+for handler, dct in handlers:
+    for format, klass in dct.items():
+        if format in {"CHAIN", "MEMORY", "MINIMAL", "NULL"}:
             continue  # get their own pages
-        FILE_TYPES[fmt][clstype] = klass
+        FILE_TYPES[format][handler] = klass
+
 
 sorted_types = sorted(FILE_TYPES.items())
```

Comment:
Could we make this SORTED_TYPES so it doesn't look like every other variable -- I was confused with the code lower down for a bit!